### DISCUSSION TITLE
Popover can't be initially opened when rendered inside focus-trap-react

### DISCUSSION THREAD

**Discussion by moroshko** [radix-ui/primitives#2025] [Help] [answered]
I'm rendering an initially opened `Popover` inside a modal component that uses a [`FocusTrap`](https://github.com/focus-trap/focus-trap-react).
Here is a simplified version (that just includes a `FocusTrap`, without being in a modal):
```tsx
<FocusTrap>
  <div style={{ border: "1px solid", padding: "16px" }}>
    <h2>FocusTrap</h2>
    <input type="text" />
    <Popover.Root
      defaultOpen
      onOpenChange={(open) => {
        console.log("onOpenChange", open);
      }}
    >
      <Popover.Trigger asChild>
        <button>Filter</button>
      </Popover.Trigger>
      <Popover.Portal>
        <Popover.Content>Popover content</Popover.Content>
      </Popover.Portal>
    </Popover.Root>
  </div>
</FocusTrap>
```
The issue is that `Popover.Root` is calling `onOpenChange` with `open` being `false`, which causes the `Popover` to close and not be open initially as desired.

[CodeSandbox](https://codesandbox.io/s/magical-johnson-37qq4b?file=/src/App.tsx:170-758)

My gut feel is that `FocusTrap` is trapping the focus and not allowing the `Popover` to have the focus when initially opened, but I'm not sure how best to deal with this.

Any ideas?

**Comment by andy-hook** [comment, marked as answer]
Hi @moroshko , `FocusTrap` is stealing focus from the popover after it initially opens, triggering the built in "close when focused outside" behaviour.

Given your example code, one option is to disable `FocusTrap` while `Popover` is open:
https://codesandbox.io/s/focus-trap-popover-initial-open-drwk3u?file=/src/App.tsx

You can also call `event.preventDefault` in `onFocusOutside` on `Popover.Content` to prevent the default closing behaviour.

Finally, depending on your use case you could forego `FocusTrap` entirely and instead use a Radix dialog with `Modal` set to `true`.

Hope this helps!
